stdlib/obj.ml

```diff
@@ -74,6 +74,33 @@ let int_tag = 1000
 let out_of_heap_tag = 1001
 let unaligned_tag = 1002
 
+module Closure = struct
+  type info = {
+    arity: int;
+    start_env: int;
+  }
+
+  let info_of_raw (info : nativeint) =
+    let open Nativeint in
+    (* the nativeints below are unsigned, but we know they can
+       always fit an OCaml integer so we use [to_int]
+       instead of [unsigned_to_int]. *)
+    let arity =
+      if Sys.word_size = 64 then
+        to_int (shift_right_logical info 56)
+      else
+        to_int (shift_right_logical info 24)
```

Comment:
Arity is signed (negative arity = tupled function), so please use `shift_right` and update the comment.

Reply:
Thanks! This should now be fixed.